include/xtensor/xadapt.hpp

```diff
@@ -100,7 +100,7 @@ namespace xt
      */
     template <class C, std::size_t N>
     xtensor_adaptor<C, N, layout_type::dynamic>
-    xadapt(C& container, const std::array<typename C::size_type, N>& shape, const std::array<typename C::size_type, N>& strides);
+    xadapt(C& container, const stat_shape<typename C::size_type, N>& shape, const stat_shape<typename C::size_type, N>& strides);
```

Comment:
On that note, I also think we shouldn't use the `size_t` template everywhere. It would be much better to define the `xshape<xt::index_t>` somewhere and then have it consistent across the library and only one place to change it.